File: Medium/3021. Alice and Bob Playing Flower Game/solution.py
class Solution:
    def flowerGame(self, n: int, m: int) -> int:
        num_of_odd_in_n = (n - 1) // 2 + 1
        num_of_even_in_m = (m - 2) // 2 + 1 if m > 1 else 0
        return (num_of_odd_in_n * num_of_even_in_m) + (n - num_of_odd_in_n) * (m - num_of_even_in_m)

File: Medium/3021. Alice and Bob Playing Flower Game/solution.cs
public class Solution {
    public long FlowerGame(int n, int m) {
        long num_odd_in_n = (n - 1) / 2 + 1;
        long num_even_in_m = m > 1 ? (m - 2) / 2 + 1 : 0;
        return num_odd_in_n * num_even_in_m + (n - num_odd_in_n) * (m - num_even_in_m);
    }
}